src/registry/generate.rs

```diff
@@ -73,14 +75,20 @@ pub(crate) fn command(
     )
     .expect("Failed to create template engine");
 
-    engine
-        .generate_registry(
-            logger.clone(),
-            &schema.registries[0],
-            &schema.catalog,
-            args.output.as_path(),
-        )
-        .expect("Failed to generate registry assets");
+    let template_registry =
+        TemplateRegistry::try_from_resolved_registry(&schema.registries[0], &schema.catalog)
```

Comment:
I'm not a huge fun of always indexing the first registry here.  Did you have a bug open to clean that up at some point or alter the representation?

I think you could even just pass the entire `&schema` directly to `try_from_resolved_registry` to hide that complexity until we clean it up.

Reply:
I will open an issue to improve this as it's indeed far from optimal. It was like an intermediary/unfinished step, the final goal being at some point to support multiple registries from the same app telemetry schema (e.g. OTel official registry + 0 to n vendor or internal registries).